src/lazy/encoder/writer.rs

```diff
@@ -92,6 +96,46 @@ impl<E: Encoding, Output: Write> Writer<E, Output> {
         Ok(writer)
     }
 
+    pub fn compile_macro(&mut self, source: impl IonInput) -> IonResult<Macro> {
+        let mut reader = Reader::new(AnyEncoding, source)?;
```

Comment:
🪧 Eventually I'd like to optimize this by either accepting bulk macro definitions, re-using the `Reader`, or just offering a programmatic API for constructing a macro.